packages/block-library/src/block/edit.js

```diff
@@ -98,11 +99,24 @@ function hasOverridableBlocks( blocks ) {
 }
 
 function getOverridableAttributes( block ) {
-	return Object.entries( block.attributes.metadata.bindings )
-		.filter(
-			( [ , binding ] ) => binding.source === 'core/pattern-overrides'
-		)
-		.map( ( [ attributeKey ] ) => attributeKey );
+	const set = new Set();
+	for ( const [ attributeKey, binding ] of Object.entries(
+		block.attributes.metadata.bindings
+	) ) {
+		if (
+			attributeKey === '__default' &&
+			binding.source === 'core/pattern-overrides'
+		) {
+			PARTIAL_SYNCING_SUPPORTED_BLOCKS[ block.name ].forEach(
+				( attribute ) => {
+					set.add( attribute );
+				}
+			);
+		} else if ( binding.source === 'core/pattern-overrides' ) {
+			set.add( attributeKey );
+		}
```

Comment:
It might be good to support this kind of use case in the bindings attribute:
```js
{ bindings: { __default: 'core/pattern-overrides', someOtherAttribute: 'core/some-other-binding' } }
```

So that `__default` can be overriden by another source on a per attribute basis.

I guess there are a few ways to achieve that in this function, I leave it up to you!

Some unit tests might be a good way to assert that the function works correctly.